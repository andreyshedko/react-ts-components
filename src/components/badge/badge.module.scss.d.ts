export const badge: string;
export const body: string;
export const info: string;
export const warning: string;
export const danger: string;
