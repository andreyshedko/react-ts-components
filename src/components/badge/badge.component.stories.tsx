import React from 'react';
import { actions } from '@storybook/addon-actions';

import { BadgeComponent } from './badge.component';

export default {
    component: BadgeComponent,
    title: 'Badge Component',
};

const eventsFromNames = actions('onClick', 'onKeyDown');

export const severityInfo = () =>
    <BadgeComponent
        severity={"info"}
        text={""}
        {...eventsFromNames} />;

export const severityWarning = () =>
    <BadgeComponent
        severity={"warning"}
        text={""}
        {...eventsFromNames} />

export const severityDanger = () =>
    <BadgeComponent
        severity={"danger"}
        text={""}
        {...eventsFromNames} />