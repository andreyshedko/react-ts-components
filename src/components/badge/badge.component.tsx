import * as React from "react";
import { NavLink } from 'react-router-dom';

import styles from './badge.module.scss';

export interface BadgeComponentProps {
  url: string;
  bodyText: string;
  severity: string;
}

export const BadgeComponent = (props: BadgeComponentProps) => (
  <NavLink
    className={[styles.badge , styles[`badge_${props.severity}`]].join(' ')}
    to={props.url}>
    <section className={styles.badge_body}>
      {props.bodyText}
    </section>
  </NavLink>
);