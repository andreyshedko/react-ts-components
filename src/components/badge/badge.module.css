@import "../../variables.css";

.badge {
    padding: var(--sm);
    border-radius: 3px;
    background-color: var(--warning);
    cursor: pointer;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-family: "OpenSans", sans-serif;
    font-size: 2rem;
    user-select: none;
}

.badge .body {
    color: var(--default);
}
.warning {
    background-color: var(--warning);
}
.danger {
    background-color: var(--danger);
}
.info {
    background-color: var(--info);
}
