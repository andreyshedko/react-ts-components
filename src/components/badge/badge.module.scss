@import '../../variables';

.badge {
    padding: $sm;
    border-radius: 2px;
    background-color: $warning;
    text-decoration: none;
    &_body {
        color: $default;
    }
    &_warning {
        background-color: $warning;
    }
    &_danger {
        background-color: $danger;
    }
    &_info {
        background-color: $info;
    }
}