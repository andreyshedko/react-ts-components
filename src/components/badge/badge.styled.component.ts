import styled from "styled-components";

type MainProps = {
    readonly severity: 'info' | 'warning' | 'danger',
}

const severity = new Map([
    ['info', '#007bff'],
    ['warning', '#ff9f1c'],
    ['danger', '#e63946']]
)

export const Main = styled.main<MainProps>`
    padding: 1rem;
    border-radius: 3px;
    cursor: pointer;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-family: "OpenSans", sans-serif;
    font-size: 2rem;
    user-select: none;
    background-color: ${props => severity.get(props.severity)};
`;

export const Body = styled.section`
    color: #fff;
`;