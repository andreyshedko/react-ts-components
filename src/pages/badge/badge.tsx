import * as React from "react";
import { BadgeComponent } from "../../components/badge/badge.component";

export const BadgeInfoComponent = () => {
    return (
        <div className={['container', 'col-3'].join(' ')}>
            <BadgeComponent url="/" bodyText="Severity info example" severity={"info"} />
            <BadgeComponent url="/" bodyText="Severity warning example" severity={"warning"} />
            <BadgeComponent url="/" bodyText="Severity danger example" severity={"danger"} />
        </div>)
};