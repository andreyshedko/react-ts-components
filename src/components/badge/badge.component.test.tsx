import React from "react";

import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { BadgeComponent } from "./badge.component";

configure({ adapter: new Adapter() });

const mockFn = jest.fn();

describe("Button Component", () => {
  test("should display info severity state", async () => {
    const component = renderer.create(
      <BadgeComponent
        severity={"info"}
        text={"test"}
        onClick={mockFn}
        onKeyDown={mockFn} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display warning severity state", async () => {
    const component = renderer.create(
      <BadgeComponent
        severity={"warning"}
        text={"test"}
        onClick={mockFn}
        onKeyDown={mockFn} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display danger severity state", async () => {
    const component = renderer.create(
      <BadgeComponent
        severity={"danger"}
        text={"test"}
        onClick={jest.fn()}
        onKeyDown={jest.fn()} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("fire click event", async () => {
    const wrapper = shallow(<BadgeComponent
      severity={"danger"}
      text={"test"}
      onClick={mockFn}
      onKeyDown={mockFn} />);
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  test("fire keydown event", async () => {
    const wrapper = shallow(<BadgeComponent
      severity={"danger"}
      text={"test"}
      onClick={mockFn}
      onKeyDown={mockFn} />);
    wrapper.simulate('keydown');
    expect(mockFn).toHaveBeenCalled();
  });

}); 