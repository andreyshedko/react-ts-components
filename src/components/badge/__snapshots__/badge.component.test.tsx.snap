// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should display danger severity state 1`] = `
<main
  className="sc-AxjAm oHVtP"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <section
    className="sc-AxirZ ihGzgH"
  >
    danger
  </section>
</main>
`;

exports[`Button Component should display info severity state 1`] = `
<main
  className="sc-AxjAm bFEgVV"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <section
    className="sc-AxirZ ihGzgH"
  >
    info
  </section>
</main>
`;

exports[`Button Component should display warning severity state 1`] = `
<main
  className="sc-AxjAm jdLlxh"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <section
    className="sc-AxirZ ihGzgH"
  >
    warning
  </section>
</main>
`;
